src/test/test_connector_redis.cpp

```diff
@@ -0,0 +1,348 @@
+}
+
+#define RAND_FIRST "fa37JncCHryDsbza"
+#define RAND_SECOND "yy4cBWDxS22JjzhM"
+
+TEST_CASE_METHOD(F, "test_redis_client_ret_queue", TEST_MODULE)
+{
+    redisReply * reply;
+    string retqueue;
+
+    // Check correct behaviour from a known point
+    srand(0);
+    GetReturnQueue(con, TEST_QUEUE, retqueue);
+    CHECK(retqueue == TEST_QUEUE "_" RAND_FIRST);
+
+    GetReturnQueue(con, TEST_QUEUE, retqueue);
+    CHECK(retqueue == TEST_QUEUE "_" RAND_SECOND);
+
+    // Check behaviour when queue already exists
+    reply = (redisReply *) redisCommand(con, "LPUSH %s %s", TEST_QUEUE "_" RAND_FIRST, "test");
+    freeReplyObject(reply);
+
+    srand(0);
+    GetReturnQueue(con, TEST_QUEUE, retqueue);
+    CHECK(retqueue == TEST_QUEUE "_" RAND_SECOND);
+
+    // Check for a failure with the redis context
+    stop_redis_server(server_pid);
+    CHECK_EXCEPTION_TYPE(GetReturnQueue(con, TEST_QUEUE, retqueue), JsonRpcException, check_exception1);
+}
+
+
+TEST_CASE_METHOD(F, "test_redis_client_set_queue", TEST_MODULE)
+{
+    RedisServer server2(TEST_HOST, TEST_PORT, TEST_QUEUE "_other");
+    MockClientConnectionHandler handler2;
+    server2.SetHandler(&handler2);
+    server2.StartListening();
+
+    client.SetQueue(TEST_QUEUE "_other");
+
+    handler2.response = "exampleresponse";
+    string result;
+    client.SendRPCMessage("examplerequest", result);
+
+    CHECK(handler2.request == "examplerequest");
+    CHECK(result == "exampleresponse");
+
+    server2.StopListening();
+}
+
+
+TEST_CASE_METHOD(F, "test_redis_server_error", TEST_MODULE)
+{
+    RedisServer server2(TEST_HOST, TEST_PORT+1, TEST_QUEUE);
+
+    CHECK(server2.StartListening() == false);
+    CHECK(server.StartListening() == true);
+    CHECK(server2.StopListening() == true);
+    CHECK(server.StopListening() == true);
+    CHECK(server.StopListening() == true);
+}
+
+
+TEST_CASE_METHOD(F, "test_redis_server_bad_request", TEST_MODULE)
+{
+    string queue = TEST_QUEUE;
+    string req1 = "test";
+
+    redisReply * reply;
+
+    reply = (redisReply *) redisCommand(con, "LPUSH %s %s", queue.c_str(), req1.c_str());
+    CHECK(handler.request == "");
+    freeReplyObject(reply);
+    reply = (redisReply *) redisCommand(con, "DEL %s", queue.c_str());
+    freeReplyObject(reply);
+
+    reply = (redisReply *) redisCommand(con, "SET %s %s", queue.c_str(), req1.c_str());
+    CHECK(handler.request == "");
+    freeReplyObject(reply);
+    reply = (redisReply *) redisCommand(con, "DEL %s", queue.c_str());
+    freeReplyObject(reply);
+}
+
+
+TEST_CASE_METHOD(F, "test_redis_client_timeout", TEST_MODULE)
+{
+    RedisClient client2(TEST_HOST, TEST_PORT, "invalid_queue");
+    client2.SetTimeout(1);
+
+    string result;
+    CHECK_EXCEPTION_TYPE(client2.SendRPCMessage("examplerequest", result), JsonRpcException, check_exception1);
+}
+
+
+TEST_CASE_METHOD(F, "test_redis_server_longpost", TEST_MODULE)
```

Comment:
This is not a post request, please rename the test to `test_redis_server_longrequest`.